src/tests/Common/CoreCLRTestLibrary/Utilities.cs

```diff
@@ -488,5 +488,35 @@ public static void ExecuteAndUnload(string assemblyPath, string typeName, string
 
             Assert.False(alcWeakRef.IsAlive);
         }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct RLimit
+        {
+            public ulong rlim_cur; // Soft limit
+            public ulong rlim_max; // Hard limit
```

Comment:
If I'm understanding this definition correctly, this will be invalid on 32-bit machines, right? I think we should consider making this a `CULong` instead. I'd like to make this not be a latent issue if we run these tests on a 32-bit non-Windows machine.

Reply:
https://man7.org/linux/man-pages/man2/setrlimit.2.html says that it is 64-bit value even on 32-bit systems, so the definition using `ulong` looks correct to me.

Reply:
I'm not fully convinced. OpenBSD is defined as `int` - https://github.com/openbsd/src/blob/39200c1893dd71d26dd98a6230435389e2e9e665/gnu/gcc/gcc/sys-types.h#L131 . I see the linux kernel does define it as `__kernel_ulong_t`, which does appear to be `long long` on x86 - https://github.com/torvalds/linux/blob/5fc31936081919a8572a3d644f3fbb258038f337/include/uapi/linux/resource.h#L43-L46. Apple only defines it as 64-bit too, but I suppose that is because they only support 64-bit - https://developer.apple.com/documentation/kernel/rlim_t?changes=l__3__8.

Looks like `musl` as well - https://git.musl-libc.org/cgit/musl/tree/include/sys/resource.h#n20

The referenced documentation does seem to also indicate that so I'm okay if we want to go with it.

Reply:
We can enable this for Linux and macOS only (the only platforms where we actually need it).